back/app/services/side_fx_custom_field_service.rb

```diff
@@ -32,4 +35,24 @@ def after_destroy(frozen_custom_field, current_user)
       payload: { custom_field: serialized_custom_field }
     )
   end
+
+  private
+
+  def log_user_data_that_will_be_deleted(custom_field, current_user)
+    related_user_data = {}
+    User.where("custom_field_values ? '#{custom_field.key}'").each do |user|
+      related_user_data[user.id] = { custom_field.key.to_s => user.custom_field_values[custom_field.key] }
+    end
```

Comment:
I benchmarked this iteration. It takes about 1/100th second to process 1000 users.